### PR TITLE
`azurerm_search_service` - fix update function to adjust for changed API behavior

### LINKED ISSUE #24623: azurerm_search_service create/update fails to update because on invalid properties
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.4.6

### AzureRM Provider Version

3.57.0

### Affected Resource(s)/Data Source(s)

azurerm_search_service

### Terraform Configuration Files

```hcl
resource "azurerm_search_service" "search_service" {
  name                          = "srch-test"
  resource_group_name           = "rg-test"
  location                      = "northcentralus"
  sku                           = var.sku
  replica_count                 = var.replica_count
  partition_count               = lower(var.sku) == "basic" ? 1 : var.partition_count
  public_network_access_enabled = false
  local_authentication_enabled  = true
}
```

### Debug Output/Panic Output

```shell
Error: updating Search Service (Subscription: "a3f35929-499d-49ee-a21d-0fbc494461b3" Resource Group Name: "rg-openai-001" Search Service Name: "srch-idpaidev-001"): performing CreateOrUpdate: unexpected status 400 with error: BadRequest: Request body cannot contain the following invalid Properties : privateEndpointConnections,provisioningState,sharedPrivateLinkResources,status,statusDetails. Please refer to https://learn.microsoft.com/en-us/rest/api/searchmanagement/management-api-versions for a list of allowed properties
```

### Expected Behaviour

expect the search service local auth to be enabled.

### Actual Behaviour

Error: updating Search Service (Subscription: "a3f35929-499d-49ee-a21d-0fbc494461b3" Resource Group Name: "rg-openai-001" Search Service Name: "srch-idpaidev-001"): performing CreateOrUpdate: unexpected status 400 with error: BadRequest: Request body cannot contain the following invalid Properties : privateEndpointConnections,provisioningState,sharedPrivateLinkResources,status,statusDetails. Please refer to https://learn.microsoft.com/en-us/rest/api/searchmanagement/management-api-versions for a list of allowed properties

### Steps to Reproduce

1. Create a search service first with local auth set to false.
2. then change the local auth to true. and apply.

### Important Factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by WodansSon**
(fixes #24623)

### CODE REVIEW

File: internal/services/search/search_service_resource.go [hunk 1]
```diff
@@ -341,20 +341,39 @@ func resourceSearchServiceUpdate(d *pluginsdk.ResourceData, meta interface{}) er
...
+	// status and statusDetails)
+	payload := services.SearchService{
+		Id:         model.Id,
+		Identity:   model.Identity,
+		Location:   model.Location,
+		Name:       model.Name,
+		Properties: pointer.To(services.SearchServiceProperties{}),
+		Sku:        model.Sku,
+		Tags:       model.Tags,
+		Type:       model.Type,
```
Comment (tombuildsstuff) [inline internal/services/search/search_service_resource.go:367]: Can we remove these three properties? These fields should (legacy APIs aside) never need to be set in the root of the payload:

```suggestion
		Identity:   model.Identity,
		Location:   model.Location,
		Properties: pointer.To(services.SearchServiceProperties{}),
		Sku:        model.Sku,
		Tags:       model.Tags,
```

**Review by tombuildsstuff — APPROVED** [review]
One comment about removing `Id`, `Name` and `Type` (since these should _almost_ never be set in the root payload, per the ARM Spec - with some APIs reject these) - but this otherwise LGTM 👍

File: internal/services/search/search_service_resource.go (hunk 1 above)
Comment (WodansSon) [reply internal/services/search/search_service_resource.go]: Fixed.

**Comment by WodansSon** [comment]
image

**Pull request merged**